packages/flutter/lib/src/widgets/linked_text.dart

```diff
@@ -0,0 +1,776 @@
+  /// See also:
+  ///
+  ///  * [LinkedText.regExp], which automatically finds ranges that match
+  ///    the given [RegExp].
+  ///  * [LinkedText.textLinkers], which uses [TextLinker]s to allow full
+  ///    control over matching and building different types of links.
+  LinkedText({
+    super.key,
+    this.builder = defaultBuilder,
+    required LinkTapCallback onTap,
+    List<InlineSpan>? spans,
+    String? text,
+  }) : assert((text == null) != (spans == null), 'Must specify exactly one to link: either text or spans.'),
+       textLinkers = <TextLinker>[
+         TextLinker(
+           textRangesFinder: defaultTextRangesFinder,
+           linkBuilder: getDefaultLinkBuilder(onTap),
+         ),
+       ],
+       spans = spans ?? <InlineSpan>[
+         TextSpan(
+           text: text,
+         ),
+       ];
+
+  /// Creates an instance of [LinkedText] where text matched by the given
+  /// [RegExp] is made interactive.
+  ///
+  /// The [RegExp] must not produce overlapping matches.
+  ///
+  /// {@tool dartpad}
+  /// This example shows how to use [LinkedText.regExp] to link Twitter handles.
+  ///
+  /// ** See code in examples/api/lib/widgets/linked_text/linked_text.1.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [LinkedText.new], which can be passed [TextRange]s directly or
+  ///    otherwise matches URLs by default.
+  ///  * [LinkedText.textLinkers], which uses [TextLinker]s to allow full
+  ///    control over matching and building different types of links.
+  LinkedText.regExp({
+    super.key,
+    this.builder = defaultBuilder,
+    required LinkTapCallback onTap,
+    required RegExp regExp,
+    List<InlineSpan>? spans,
+    String? text,
+  }) : assert((text == null) != (spans == null), 'Must specify exactly one to link: either text or spans.'),
+       textLinkers = <TextLinker>[
+         TextLinker(
+           textRangesFinder: TextLinker.textRangesFinderFromRegExp(regExp),
+           linkBuilder: getDefaultLinkBuilder(onTap),
+         ),
+       ],
+       spans = spans ?? <InlineSpan>[
+         TextSpan(
+           text: text,
+         ),
+       ];
+
+  /// Creates an instance of [LinkedText] where the given [textLinkers] are
+  /// applied.
+  ///
+  /// {@template flutter.widgets.LinkedText.textLinkers}
+  /// Useful for independently matching different types of strings with
+  /// different behaviors. For example, highlighting both URLs and Twitter
+  /// handles with different style and/or behavior.
+  /// {@endtemplate}
+  ///
+  /// The [TextLinker.textRangesFinder] must not produce overlapping
+  /// [TextRange]s.
+  ///
+  /// {@tool dartpad}
+  /// This example shows how to use [LinkedText.textLinkers] to link both URLs
+  /// and Twitter handles independently.
+  ///
+  /// ** See code in examples/api/lib/widgets/linked_text/linked_text.2.dart **
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [LinkedText.new], which can be passed [TextRange]s directly or
```

Comment:
Where is the `LinkedText.new` constructor? Not sure I see it.

Reply:
Ahh I see `LinkedText == LinkedText.new` this is the first time I see it referenced like this.

Reply:
Yeah `[LinkedText.new]` references the default constructor.  If I just reference `[LinkedText]` then it will go to the class, not the constructor.  I also just learned this recently haha.